File: lab3/app/src/main/java/com/example/lab3_thread/MainActivity.java
package com.example.lab3_thread;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnLength = findViewById(R.id.btnLength);
        Button btnWeight = findViewById(R.id.btnWeight);
        Button btnTemperature = findViewById(R.id.btnTemperature);

        btnLength.setOnClickListener(v -> openConverter("length"));
        btnWeight.setOnClickListener(v -> openConverter("weight"));
        btnTemperature.setOnClickListener(v -> openConverter("temperature"));
    }

    private void openConverter(String category) {
        Intent intent = new Intent(this, ConverterActivity.class);
        intent.putExtra("CATEGORY", category);
        startActivity(intent);
    }
}

File: variant/app/src/main/java/com/example/variant/MainActivity.kt
package com.example.variant

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val surnameEdit = findViewById<EditText>(R.id.surnameEdit)
        val nameEdit = findViewById<EditText>(R.id.nameEdit)
        val groupEdit = findViewById<EditText>(R.id.groupEdit)

        val calcButton = findViewById<Button>(R.id.calcButton)
        val doneButton = findViewById<Button>(R.id.doneButton)
        val resultText = findViewById<TextView>(R.id.resultText)

        calcButton.setOnClickListener {
            val surname = surnameEdit.text.toString()
            val name = nameEdit.text.toString()
            val group = groupEdit.text.toString()

            val sumCodes = (surname + name + group).sumOf { it.code }
            val variant = (sumCodes % 10) + 1

            resultText.text =
                "$surname\n$name\n$group\n$variant"
        }

        doneButton.setOnClickListener {
            finish()
        }
    }
}